### PR TITLE
Read storefront renderer token from Environment

### LINKED ISSUE #545: [Bug]: Shopify theme dev "using credentials that do not work with shopify theme serve"
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

Theme

### Expected behavior

Local Theme should have been served

### Actual behavior

Got the error:
"It looks like you are using credentials that do not work with shopify theme serve."

Try this ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
┃ Run shopify logout and shopify login --password "" --store STORE to force the authentication thought your browser.

### Stack trace

_No response_

### Reproduction steps

1. Run shopify theme dev --store mystore.myshopify.com
2. Error appears

### Operating System

Windows 11

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.15

### Shell

Git Bash

### Node version (run `node -v` if you're not sure)

16.15.1

### What language and version are you using in your application?

N/A

### LINKED ISSUE #601: [Bug]: Stopped working suddenly
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

Theme

### Expected behavior

Hi,

I am not sure why, but suddenly, Shopify CLI v3 stopped working completely.

Whenever I enter the shopify theme dev command, I face this error:

image

The odd thing here is ... that I no longer have the v2 installed on my system, and the shopify logout command no longer exists in v3, so having a v3 command returning me  is strange.

### Actual behavior

I really do not know what or how it happened, the v3 worked without any for those last two weeks, and suddenly, it became unusable :/.

### Stack trace

_No response_

### Reproduction steps

1. Just launch a v3 command.
2. Receive what seems to be a v2 command
3.

### Operating System

OS X

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.16

### Shell

_No response_

### Node version (run `node -v` if you're not sure)

_No response_

### What language and version are you using in your application?

_No response_

### PR DESCRIPTION
**Pull request by isaacroldan**
### WHY are these changes introduced?
When using the CLI2 from CLI3, tokens are passed via Environment and not stored in DB. 
So we need to try to get the token from the Environment first, this was causing a weird edge case for some theme users.

Fixes https://github.com/Shopify/cli/issues/601
Fixes https://github.com/Shopify/cli/issues/545

### WHAT is this pull request doing?
- In serve.rb, read the storefront token first from Environment

### Update checklist

- [ ] I've added a CHANGELOG entry for this PR (if the change is public-facing)
- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows).
- [ ] I've left the version number as is (we'll handle incrementing this when releasing).
- [ ] I've included any post-release steps in the section above (if needed).

### CODE REVIEW

File: lib/project_types/theme/commands/serve.rb [hunk 1]
```diff
@@ -77,6 +77,7 @@ def exchange_token
       end
 
       def storefront_renderer_token
+        Environment.storefront_renderer_auth_token || 
         ShopifyCLI::DB.get(:storefront_renderer_production_exchange_token)
```
Comment (karreiro) [inline lib/project_types/theme/commands/serve.rb:81]: I believe we need some extra spaces here to solve the Rubocop issue:
```suggestion
        Environment.storefront_renderer_auth_token || 
          ShopifyCLI::DB.get(:storefront_renderer_production_exchange_token)
```

**Review by karreiro — COMMENTED** [review]
Thank you, @isaacroldan! I've left only one minor comment :)

**Comment by bakura10** [comment]
Until this is merged, is there a known workaround? I am unfortunately currently stuck and cannot do any progress as it seems my Shopify CLI entered in a broken mode :/

**Comment by isaacroldan** [comment]
@bakura10 Not sure there is a valid workaround 🤔 
Did you tried logging out and in again?

We'll release a new version of the CLI today anyway! I'll make sure this is included

**Review by matteodepalo — APPROVED** [review]

**Pull request merged**